VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs

```diff
@@ -74,6 +81,27 @@ public async Task<IActionResult> Exchange(OpenIdConnectRequest request)
 
                 // Create a new authentication ticket.
                 var ticket = await CreateTicketAsync(request, user);
+                var claims = await _userClaimsPrincipalFactory.CreateAsync(user);
+                var limitedPermissions = _authorizationOptions.LimitedCookiePermissions?.Split(';', StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
+
+                if (!user.Roles.Select(r => r.Name).Contains(PlatformConstants.Security.Roles.Administrator))
+                {
+                    limitedPermissions = claims.Claims.Where(c => c.Type == "permission").Select(c => c.Value).Intersect(limitedPermissions, StringComparer.OrdinalIgnoreCase).ToArray();
```

Comment:
The `"permission"` string here should probably be replaced by the `PlatformConstants.Security.Claims.PermissionClaimType` constant.